bot/exts/moderation/voice_gate.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from contextlib import suppress
+from datetime import datetime, timedelta
+
+import discord
+from dateutil import parser
+from discord import Colour
+from discord.ext.commands import Cog, Context, command
+
+from bot.api import ResponseCodeError
+from bot.bot import Bot
+from bot.constants import Channels, Event, MODERATION_ROLES, Roles, VoiceGate as GateConf
+from bot.decorators import has_no_roles, in_whitelist
+from bot.exts.moderation.modlog import ModLog
+from bot.utils.checks import InWhitelistCheckFailure
+
+log = logging.getLogger(__name__)
+
+FAILED_MESSAGE = """{user} you don't meet with our current requirements to pass Voice Gate. You {reasons}."""
+
+MESSAGE_FIELD_MAP = {
+    "verified_at": f"haven't been verified for at least {GateConf.minimum_days_verified} days",
+    "voice_banned": "are voice banned",
+    "total_messages": f"haven't sent at least {GateConf.minimum_messages} messages",
+}
+
+
+class VoiceGate(Cog):
+    """Voice channels verification management."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @property
+    def mod_log(self) -> ModLog:
+        """Get the currently loaded ModLog cog instance."""
+        return self.bot.get_cog("ModLog")
+
+    @command(aliases=('voiceverify', 'vverify', 'voicev', 'vv'))
+    @has_no_roles(Roles.voice_verified)
+    @in_whitelist(channels=(Channels.voice_gate,), redirect=None)
+    async def voice_verify(self, ctx: Context, *_) -> None:
+        """
+        Apply to be able to use voice within the Discord server.
+
+        In order to use voice you must meet all three of the following criteria:
+        - You must have over a certain number of messages within the Discord server
+        - You must have accepted our rules over a certain number of days ago
+        - You must not be actively banned from using our voice channels
+        """
+        # Send this as first thing in order to return after sending DM
+        await ctx.send("You will get response to DM.")
+
+        try:
+            data = await self.bot.api_client.get(f"bot/users/{ctx.author.id}/metricity_data")
+        except ResponseCodeError as e:
+            if e.status == 404:
+                embed = discord.Embed(
+                    title="Not found",
+                    description=f"{ctx.author.mention} Unable to find Metricity data about you.",
```

Comment:
More appropriate DM description.
```suggestion
                    description=f"We were unable to find user data for you. Please try again shortly, if this problem persists please contact the server staff through Modmail.",
```